.github/workflows/build_and_deploy_documentation.yml

```diff
@@ -0,0 +1,36 @@
+name: "Build Sphinx Documentation"
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main 
+      - ci/setup
+
+permissions:
+  contents: write
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install dependencies
+      run: |
+        pip install myst-parser
+        pip install .
+        pip install sphinx-rtd-theme sphinx-autodoc-typehints sphinx-click sphinx-automodapi texext
+    - name: "Parse into HTML"
+      run: |
+        sphinx-apidoc -o docs/source/api src/modalities
+        sphinx-build -M html docs/source/ docs/build/
+    - name: Deploy to GitHub Pages
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        publish_branch: gh-pages
```

Comment:
Does this mean, that the content is published to this branch? 
If so, isn't there the chance for conflicts when storing the documention on  parallel branch to main?

Reply:
okay I found your comment about this, that the `gh-pages` branch "listens" to `main`. What do you mean by that?

Reply:
The Github action workflow listens to main and pushes the HTML files to the gh-pages branch.

Reply:
We store the raw rst files on main, but not the HTML files. These will be created by sphinx and pushed only to the gh-pages branch